File: unit6/java/project/Cipher.java
import java.util.Scanner;

public class Cipher {

    public static final int WIDTH = 200;
    public static final int HEIGHT = 200;
    public static final int CHANNELS = 3;
    public static final int MAX_COLOR_VAL = 255;

    public static int[][][] PIXELS = new int[HEIGHT][WIDTH][CHANNELS];

    public static void encodeMessage(String message) {
        // TODO
    }

    public static String decodeMessage() {
        String message = "";
        // TODO
        return message;
    }

    public static void main(String[] args) {
        boolean encode;
        String infile;
        String outfile = "";
        String message = "";
        Scanner scan = new Scanner(System.in);
        System.out.print("Do you want to decode (0) or encode (1)? ");
        encode = Integer.parseInt(scan.nextLine()) == 1;
        System.out.print("Select input image: ");
        infile = scan.nextLine();
        PIXELS = ImageHelper.readImage(infile);
        if (encode) {
            System.out.print("Select output image: ");
            outfile = scan.nextLine();
            System.out.println("Enter a secret sentence:");
            message = scan.nextLine();
        }
        if (encode) {
            System.out.println("Encoding secret message in image...");
            encodeMessage(message);
            ImageHelper.writeImage(PIXELS, outfile);
            System.out.println("Wrote image to destination file: " + outfile);
        } else {
            System.out.println("Decoding secret message from image...");
            message = decodeMessage();
            System.out.println("The secret message is: ");
            System.out.println(message);
        }
    }

}

File: unit6/python/project/cipher.py
from imagehelper import read_image, write_image


WIDTH = 200
HEIGHT = 200
CHANNELS = 3
MAX_COLOR_VAL = 255

PIXELS = [[[0, 0, 0] for x in range(WIDTH)] for y in range(HEIGHT)]


def encode_message(message: str) -> None:
    global PIXELS
    # TODO
    pass


def decode_message() -> str:
    global PIXELS
    message = ""
    # TODO
    return message


def main():
    global PIXELS
    encode = int(input("Do you want to decode (0) or encode (1)? ")) == 1
    infile = input("Select input image: ")
    outfile = None
    message = None
    PIXELS = read_image(infile)
    if encode:
        outfile = input("Select output image: ")
        message = input("Enter a secret sentence:\n")
    if encode:
        print("Encoding secret message in image...")
        encode_message(message)
        write_image(PIXELS, outfile)
        print("Wrote image to destination file: {}".format(outfile))
    else:
        print("Decoding secret message from image...")
        message = decode_message()
        print("The secret message is: ")
        print(message)


if __name__ == "__main__":
    main()
